lib/ui/widgets/song_thumbnail.dart

```diff
@@ -114,3 +129,49 @@ class PlayingSongThumbnail extends StatelessWidget {
     );
   }
 }
+
+class BufferingSongThumbnail extends StatelessWidget {
+  final double width;
+  final double height;
+  final Song song;
+  final double borderRadius;
+
+  const BufferingSongThumbnail({
+    Key? key,
+    required this.width,
+    required this.height,
+    required this.song,
+    required this.borderRadius,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      width: width,
+      height: height,
+      child: Stack(
+        children: <Widget>[
+          SongThumbnail(song: song),
+          SizedBox(
+            width: double.infinity,
+            height: double.infinity,
+            child: DecoratedBox(
+              decoration: BoxDecoration(
+                borderRadius: BorderRadius.all(Radius.circular(borderRadius)),
+                color: Colors.black54,
+              ),
+            ),
+          ),
+          Align(
+            alignment: Alignment.center,
+            child: SizedBox(
+              child: Image.asset('assets/images/buffering-song.gif'),
```

Comment:
Instead of a gif file, we can use an animation from [Spin Kit](https://pub.dev/packages/flutter_spinkit) (already installed), for example, DoubleBounce.

Reply:
yeah. that sounds really good. we can spin kit.
we can AssetsAudioPlayerExtension  instead(via the onTap method).

I will make the necessary changes for both for these.